### PR TITLE
TLS SubjectAltName sticky buffer/v8

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/5234

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1775

Previous PR: https://github.com/OISF/suricata/pull/10772

Changes since v7:
- rebased and conflicts resolved

### CODE REVIEW

File: doc/userguide/rules/tls-keywords.rst [hunk 1]
```diff
@@ -121,6 +121,19 @@ Examples::
 to use the previous name, but it's recommended that rules be converted to use
 the new name.
 
+tls.subjectaltname
+------------------
+
+Match TLS/SSL Subject Alternative Name field.
```
Comment (jufajardini) [inline doc/userguide/rules/tls-keywords.rst:127]: Seeing the PR with the multi-buffer tests and the conversation that originated that addition, I think it'd be good to add here that this supports multiple buffers, and maybe add an example? :)

Comment (jufajardini) [reply doc/userguide/rules/tls-keywords.rst:127]: Sorry for keeping adding to this but just got curious: does this support negated content?

Comment (catenacyber) [reply doc/userguide/rules/tls-keywords.rst:127]: likely but https://redmine.openinfosecfoundation.org/issues/2224

**Comment by catenacyber** [comment]
Still adding my reference to https://github.com/OISF/suricata/pull/10462

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 20242

**Pull request closed without merging**